spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,245 @@
+---
+
+ics: 3
+title: Connection Semantics & Lifecycle
+stage: proposal
+category: ibc-core
+author: Christopher Goes <cwgoes@tendermint.com>, Juwoon Yun <joon@tendermint.com>
+created: 2019-03-07
+modified: 2019-03-07
+---
+
+## Synopsis
+
+The basis of IBC is the ability to verify in the on-chain consensus ruleset of chain `B` that a data packet received on chain `B` was correctly generated on chain `A`. This establishes a cross-chain linearity guarantee: upon validation of that packet on chain `B` we know that the packet has been executed on chain `A` and any associated logic resolved (such as assets being escrowed), and we can safely perform application logic on chain `B` (such as generating vouchers on chain `B` for the chain `A` assets which can later be redeemed with a packet in the opposite direction). 
+
+This standards document outlines the abstraction of an IBC _connection_: the state and consensus ruleset necessary to perform IBC packet verification.
+
+## Specification
+
+### Motivation
+
+- Connection = cross-chain light client state.
+- Between two chains `A` and `B`.
+- Permissionless opening / closing / updates.
+
+### Desired Properties
+
+- Permissionless channel opening / channel closing / channel updates.
+
+#### Pre-Establishment
+
+- Guarantees that no packets can be committed on other connections?
+- No required a priori root-of-trust knowledge
+- Only one connection "between" two chains
+
+#### During Handshake
+
+Once a negotiation handshake has *begun* (defined as the first packet being committed):
+
+- Only the appropriate handshake packets can be committed in order
+- No chain can masquerade as one of the handshaking chains (formalize...)
+
+#### Post-Establishment
+
+- Connection provides verification of relayed packets
+- No packets other than committed by consensus of *A* / *B* on connection *C* can be relayed
+
+### Technical Specification
+
+(detailed technical specification: syntax, semantics, sub-protocols, algorithms, data structures, etc)
+
+#### Definitions
+
+- Chain `A` is the source blockchain from which the IBC packet is sent
+- Chain `B` is the destination blockchain on which the IBC packet is received
+- `H_h` is the signed header of chain `A` at height `h` 
+- `C_h` is a subset of the consensus ruleset of chain `A` at height `h` 
+- `V_kh` is the value stored on chain `A` under key `k` at height `h` 
+- `P` is the unbonding period of chain `P`, in units of time
+- `dt(a, b)` is the time difference between events `a` and `b` 
+
+Note that of all these, only `H_h` defines a signature and is thus attributable.
+
+#### Requirements
+
+(reference ICS 2: consensus primitives)
+
+To facilitate an IBC connection, the two blockchains must provide the following proofs:
+
+1. Given a trusted `H_h` and `C_h` and an attributable update message `U_h`,  
```

Comment:
Should this include a statement that `H_h'` is later than `H_h` ?

Reply:
Also what is an update message and how is it relevant here? 

Same below, what is a change message and how is it relevant?

Reply:
This was a leftover from the old spec, deprecated in favor of definitions in ICS 2.